### PR TITLE
Feat/submit application attachments

### PR DESCRIPTION
**Pull request by dannil76**
## Explain the changes you’ve made
Lagt till stöd för att kunna bifoga dokument tillsammans med ansökan. Detta gör det möjligt för både grund- och löpande ansökan.

## Explain why these changes are made
Sökande ska kunna bifoga dokument tillsammans med sin ansökan för att det ska vara lätt att göra rätt

## Explain your solution
Brutit ut logik som används både för submitCompletion och submitApplication till en helper (attachment.ts)
Lagt till attributet `attachments` till payload för post anropet till VADA

## How to test

1. Checkout this branch
2. run `sls deploy` within service folder for `viva-ms`

## Tested environments

- [x] Your personal AWS environment.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-2nw5cf2 SubmitApplication ska kunna skicka med attachments](https://app.clickup.com/t/2nw5cf2)

File: services/viva-ms/src/helpers/attachment.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+async function createAttachmentFromAnswers(
+  personalNumber: PersonalNumber,
+  answerList: CaseFormAnswer[]
+): Promise<CaseAttachment[]> {
+  const attachmentList: CaseAttachment[] = [];
+
+  for (const answer of answerList) {
+    if (!isAnswerAttachment(answer)) {
+      continue;
+    }
```
[the remark this answers is not in the document]
Comment (kajm) [reply services/viva-ms/src/helpers/attachment.ts:62]:
```js
import * as S3 from '../../src/libs/S3';

jest.mock('../../src/libs/S3');

import { createAttachmentFromAnswers } from '../../src/helpers/attachment';

it('just works', async () => {
  jest
    .spyOn(S3.default, 'getFile')
    .mockResolvedValueOnce({
      Body: 'Some body here',
      id: '1',
    })
    .mockRejectedValueOnce({
      Body: 'Another body',
      id: '2',
    });

  const result = await createAttachmentFromAnswers('19949292-1234', [
    {
      field: {
        id: 'field_id',
        tags: [],
      },
      value: [
        {
          uploadedFileName: 'uploadedFileName_1',
        },
        {
          uploadedFileName: 'uploadedFileName_2',
        },
      ],
    },
    {
      field: {
        id: 'field_id',
        tags: [],
      },
      value: 'String value, SHOULD NOT BE ADDED',
    },
  ]);

  console.log('RESULT: ', result);
  expect(result).toBe(true);
});

```

[the remark this answers is not in the document]
Comment (dannil76) [reply services/viva-ms/src/helpers/attachment.ts:62]: Will fix it asap

File: services/viva-ms/test/lambdas/submitApplications.test.ts [hunk 2]
```diff
@@ -29,8 +35,8 @@ const details = {
   },
 };
 
-let event: LambdaEvent;
-let context: LambdaContext;
+let event: LambdaRequest;
```
Comment (kajm) [inline services/viva-ms/test/lambdas/submitApplications.test.ts:38]: event -> input ?

**Review by kajm — CHANGES_REQUESTED** [review]

**Review by kajm — APPROVED** [review]

**Pull request merged**